### PR TITLE
Scroll to improvements

### PR DESCRIPTION
**Pull request by naqvitalha**
## Description

`viewPosition` and `viewOffset` support added to `scrollTo` methods.

## Checklist

- [x] I have added a changelog entry following the [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) guidelines.

### CODE REVIEW

File: src/FlashList.tsx [hunk 1]
```diff
@@ -770,15 +770,39 @@ class FlashList<T> extends React.PureComponent<
...
+    const layout = this.rlvRef?.getLayout(params.index);
+    const listSize = this.rlvRef?.getRenderedSize();
+
+    if (layout && listSize) {
+      const itemOffset = this.props.horizontal ? layout.x : layout.y;
+      const fixedDimension = this.props.horizontal
+        ? listSize.width
+        : listSize.height;
+      const itemSize = this.props.horizontal ? layout.width : layout.height;
+      const scrollOffset =
```
Comment (naqvitalha) [inline src/FlashList.tsx:782]: Exact same logic as FlatList

File: src/__tests__/FlashList.test.tsx [hunk 2]
```diff
@@ -770,4 +770,33 @@ describe("FlashList", () => {
...
+    const flashList = mountFlashList({
+      data: new Array(40).fill(1).map((_, index) => {
+        return index.toString();
+      }),
+    });
+    const plv = flashList.find(ProgressiveListView)
+      ?.instance as ProgressiveListView;
+    const scrollToOffset = jest.spyOn(plv, "scrollToOffset");
+    flashList.instance.scrollToIndex({ index: 10, viewPosition: 0.5 });
+    expect(scrollToOffset).toBeCalledWith(1650, 1650, false, true);
```
Comment (ElviraBurchik) [inline src/__tests__/FlashList.test.tsx:783]: Can we add more information about how expected values are calculated?

File: src/FlashList.tsx [hunk 3]
```diff
@@ -770,15 +770,39 @@ class FlashList<T> extends React.PureComponent<
     viewPosition?: number | undefined;
   }) {
     // known issue: no support for view offset/position
```
Comment (ElviraBurchik) [inline src/FlashList.tsx:772]:
```suggestion
```

File: src/__tests__/FlashList.test.tsx [hunk 4]
```diff
@@ -770,4 +770,33 @@ describe("FlashList", () => {
     flashList.setProps({ disableAutoLayout: true });
     expect(flashList.find(AutoLayoutView)?.props.disableAutoLayout).toBe(true);
   });
+  it("Computed correct scrollTo offset if view position/offset are specified", () => {
```
Comment (ElviraBurchik) [inline src/__tests__/FlashList.test.tsx:773]: Let's move it to two separate test cases? It will be a little easier to comprehend

**Pull request merged**